### PR TITLE
feat(analyzer): add certificate analyzer

### PR DESCRIPTION
**Pull request by DexterYan**
## Description, Motivation and Context
- this is based on the https://github.com/replicatedhq/troubleshoot/pull/1119
- add certificate analyzer and test cases

issue: #762

# Support Bundle Yaml
```
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: default
spec:
  collectors:
    - certificates: 
        secrets:
            - name: envoycert
              namespaces:
                - kube-system
                - projectcontour
        configMaps:
            - name: kube-root-ca.crt
              namespaces:
                - curlie
                - kurl
  analyzers:
    - certificates: # Iterate through list of certificates
        outcomes:
          - pass:
              message: "certificate is valid"
          - warn:
              when: "notAfter < Today + 365 days"
              message: "certificate is about to expire"
          - fail:
              when: "notAfter < Today"
              message: "certificate has expired"
```

# Result Json
```
[
    {
        "name": "cerfiticates.verification",
        "labels": {
            "desiredPosition": "1",
            "iconKey": "",
            "iconUri": ""
        },
        "insight": {
            "name": "cerfiticates.verification",
            "labels": {
                "iconKey": "",
                "iconUri": ""
            },
            "primary": "Cerfiticates Verification",
            "detail": "ca.crt certificate is about to expire in 365 days, obtained from envoycert secret within projectcontour namespace",
            "severity": "warn"
        },
        "severity": "warn",
        "analyzerSpec": ""
    },
    {
        "name": "cerfiticates.verification",
        "labels": {
            "desiredPosition": "1",
            "iconKey": "",
            "iconUri": ""
        },
        "insight": {
            "name": "cerfiticates.verification",
            "labels": {
                "iconKey": "",
                "iconUri": ""
            },
            "primary": "Cerfiticates Verification",
            "detail": "tls.crt certificate is about to expire in 365 days, obtained from envoycert secret within projectcontour namespace",
            "severity": "warn"
        },
        "severity": "warn",
        "analyzerSpec": ""
    },
    {
        "name": "cerfiticates.verification",
        "labels": {
            "desiredPosition": "1",
            "iconKey": "",
            "iconUri": ""
        },
        "insight": {
            "name": "cerfiticates.verification",
            "labels": {
                "iconKey": "",
                "iconUri": ""
            },
            "primary": "Cerfiticates Verification",
            "detail": "ca.crt certificate is valid, obtained from kube-root-ca.crt configmap within kurl namespace",
            "severity": "debug"
        },
        "severity": "debug",
        "analyzerSpec": ""
    }
]
```

## Checklist

- [x] New and existing tests pass locally with introduced changes.
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] The commit message(s) are informative and highlight any breaking changes
- [ ] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: pkg/analyze/certificates.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+)
+
+type AnalyzeCertificates struct {
+	analyzer *troubleshootv1beta2.CertificatesAnalyze
+}
+
+func (a *AnalyzeCertificates) Title() string {
+	title := a.analyzer.CheckName
+	if title == "" {
+		return "Cerfiticates Verification"
```
Comment (CpuID) [inline pkg/analyze/certificates.go]:
```suggestion
		return "Certificates Verification"
```

File: pkg/analyze/certificates_test.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+						},
+					},
+				},
+			},
+			expectResult: []*AnalyzeResult{
+				{
+					IsPass:  true,
+					IsWarn:  false,
+					IsFail:  false,
+					Title:   "Cerfiticates Verification",
```
Comment (CpuID) [inline pkg/analyze/certificates_test.go]:
```suggestion
					Title:   "Certificates Verification",
```

File: pkg/analyze/certificates_test.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+						},
+					},
+				},
+			},
+			expectResult: []*AnalyzeResult{
+				{
+					IsPass:  false,
+					IsWarn:  false,
+					IsFail:  true,
+					Title:   "Cerfiticates Verification",
```
Comment (CpuID) [inline pkg/analyze/certificates_test.go]:
```suggestion
					Title:   "Certificates Verification",
```

File: pkg/analyze/certificates_test.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+						},
+					},
+				},
+			},
+			expectResult: []*AnalyzeResult{
+				{
+					IsPass:  false,
+					IsWarn:  true,
+					IsFail:  false,
+					Title:   "Cerfiticates Verification",
```
Comment (CpuID) [inline pkg/analyze/certificates_test.go]:
```suggestion
					Title:   "Certificates Verification",
```

File: pkg/analyze/certificates.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+	collectorCertificates := []collect.CertCollection{}
+	if err := json.Unmarshal(certificatesInfo, &collectorCertificates); err != nil {
+		return nil, errors.Wrap(err, "failed to parse certificates.json")
+	}
+
+	return a.analyzeAnalyzeCertificatesResult(collectorCertificates, analyzer.Outcomes)
+}
+
+func (a *AnalyzeCertificates) analyzeAnalyzeCertificatesResult(certifcates []collect.CertCollection, outcomes []*troubleshootv1beta2.Outcome) ([]*AnalyzeResult, error) {
```
Comment (CpuID) [inline pkg/analyze/certificates.go]:
```suggestion
func (a *AnalyzeCertificates) analyzeAnalyzeCertificatesResult(certificates []collect.CertCollection, outcomes []*troubleshootv1beta2.Outcome) ([]*AnalyzeResult, error) {
```

File: pkg/analyze/certificates.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+	var results []*AnalyzeResult
+
+	for _, cert := range certifcates {
```
Comment (CpuID) [inline pkg/analyze/certificates.go]:
```suggestion
	for _, cert := range certificates {
```

**Review by CpuID — CHANGES_REQUESTED** [review]
overall LGTM, haven't tested it locally but the tests exist and are passing according to GitHub :)

just a few spelling fixes to apply 👍

**Review by CpuID — APPROVED** [review]

**Review by xavpaice — DISMISSED** [review]

**Review by xavpaice — APPROVED** [review]

**Pull request merged**